e2e/definition_test.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"github.com/mesg-foundation/engine/protobuf/api"
+	pb "github.com/mesg-foundation/engine/protobuf/api"
+	"github.com/mesg-foundation/engine/service"
+)
+
+func newTestCreateServiceRequest() *pb.CreateServiceRequest {
+	return &api.CreateServiceRequest{
+		Sid:  "test-service",
+		Name: "test-service",
+		Configuration: service.Service_Configuration{
+			Env: []string{"FOO=1", "BAR=2", "REQUIRED"},
+		},
+		Tasks: []*service.Service_Task{
+			{
+				Key: "ping",
+				Inputs: []*service.Service_Parameter{
+					{
+						Key:  "msg",
+						Type: "String",
+					},
+				},
+				Outputs: []*service.Service_Parameter{
+					{
+						Key:  "pong",
+						Type: "String",
+					},
+				},
+			},
+			{
+				Key: "add",
+				Inputs: []*service.Service_Parameter{
+					{
+						Key:  "n",
+						Type: "Number",
+					},
+					{
+						Key:  "m",
+						Type: "Number",
+					},
+				},
+				Outputs: []*service.Service_Parameter{
+					{
+						Key:  "res",
+						Type: "Number",
+					},
+				},
+			},
+			{
+				Key: "error",
+			},
+		},
+		Events: []*service.Service_Event{
+			{
+				Key: "test_service_ready",
+			},
+			{
+				Key: "ping_ok",
+				Data: []*service.Service_Parameter{
+					{
+						Key:  "msg",
+						Type: "String",
+					},
+				},
+			},
+			{
+				Key: "add_ok",
+				Data: []*service.Service_Parameter{
+					{
+						Key:  "msg",
+						Type: "String",
+					},
+				},
+			},
+			{
+				Key: "error_ok",
+				Data: []*service.Service_Parameter{
+					{
+						Key:  "msg",
+						Type: "String",
+					},
+				},
+			},
+		},
+		Source: "QmPG1Ze96pH1EgVMWsGKM33jXoG63rigMncSEqZXP7oncq",
```

Comment:
I like to not depend on the cli compilation of the service but we still need to use the cli to archive and upload the source code to ipfs.
Could this hash be at least read from a dedicated file? Or even better, the source code to be uploaded to ipfs every time? Or even, create a local http server to serve the source code when the engine require it (with a modif of the `config.ipfsendpoint`)?

Reply:
For sure that would be great, but as for now it's kind of overkill feature with questionable benefits.

For dedicated file we need to set up file coping (for local machine, ci). Same for http server.

This could be done but in for sure in differenent PR and with lower priority.